File: MyStagePass.Model/Event.cs
using System;
using System.Collections.Generic;

namespace MyStagePass.Model
{
	public class Event
	{
		public int EventID { get; set; }
		public string? EventName { get; set; }
		public string? Description { get; set; }
		public int TicketsSold { get; set; }
		public int TicketsAvailable { get; set; }
		public byte[]? Image { get; set; }
		public int PerformerID { get; set; }
		public virtual Performer Performer { get; set; } = null!;
		public DateTime EventDate { get; set; }
		public int LocationID { get; set; }
		public virtual Location Location { get; set; } = null!;
		public int StatusID { get; set; }
		public virtual Status Status { get; set; } = null!;

	}
}

File: UI/mystagepass_admin/lib/models/Event/event.dart
import 'package:json_annotation/json_annotation.dart';
import '../Performer/performer.dart';
import '../Location/location.dart';
import '../Status/status.dart';

part 'event.g.dart';

@JsonSerializable()
class Event {
  @JsonKey(name: 'eventID')
  int? eventId;
  String? eventName;
  int? ticketsSold;
  String? locationName;
  DateTime? eventDate;
  String? timeStatus;
  int? regularPrice;
  int? vipPrice;
  int? premiumPrice;
  DateTime? createdAt;
  @JsonKey(name: 'isCancelled')
  bool? isCancelled;
  Performer? performer;
  Location? location;
  Status? status;

  Event({
    this.eventId,
    this.eventName,
    this.ticketsSold,
    this.locationName,
    this.eventDate,
    this.timeStatus,
    this.regularPrice,
    this.vipPrice,
    this.premiumPrice,
    this.isCancelled,
    this.performer,
    this.location,
    this.status,
    this.createdAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);
  Map<String, dynamic> toJson() => _$EventToJson(this);
}
